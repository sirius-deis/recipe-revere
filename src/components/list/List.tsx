import React, { FC, PropsWithChildren, ReactNode } from "react";

import styles from "./List.module.css";
import ListItem from "../listItem/ListItem";

interface ListProps {
  direction: "vertical" | "horizontal";
}

const List: FC<PropsWithChildren & ListProps> = ({ children, direction }) => {
  return (
    <ul className={`${styles.list} ${styles[direction]}`}>
      {React.Children.map<ReactNode, ReactNode>(children, (child) => {
        return <ListItem>{child}</ListItem>;
      })}
    </ul>
  );
};

export default List;
