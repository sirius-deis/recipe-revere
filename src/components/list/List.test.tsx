import { render, screen } from "@testing-library/react";
import List from "./List";

const dummy = [
  <div key={1}>Some content 1</div>,
  <div key={2}>Some content 2</div>,
  <div key={3}>Some content 3</div>,
];

describe("List component", () => {
  it("should match snapshot", () => {
    const { container } = render(<List>{dummy.map((el) => el)}</List>);
    expect(container).toMatchSnapshot();
  });
  it("should add vertical direction props correctly and apply corresponding class", () => {
    render(<List direction="vertical">{dummy.map((el) => el)}</List>);
    expect(screen.getByRole("list")).toHaveClass("vertical");
  });
  it("should add horizontal direction props correctly and apply corresponding class", () => {
    render(<List direction="horizontal">{dummy.map((el) => el)}</List>);
    expect(screen.getByRole("list")).toHaveClass("horizontal");
  });
  it("should add render provided children", () => {
    render(<List direction="horizontal">{dummy.map((el) => el)}</List>);
    expect(screen.getByText("Some content 1")).toBeInTheDocument();
    expect(screen.getByText("Some content 2")).toBeInTheDocument();
    expect(screen.getByText("Some content 3")).toBeInTheDocument();
  });
});
