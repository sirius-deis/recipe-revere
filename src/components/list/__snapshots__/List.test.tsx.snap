// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should match snapshot 1`] = `
<div>
  <ul
    class="list vertical"
  >
    <li>
      <div>
        Some content 1
      </div>
    </li>
    <li>
      <div>
        Some content 2
      </div>
    </li>
    <li>
      <div>
        Some content 3
      </div>
    </li>
  </ul>
</div>
`;
